File: src_fortran/sub/geoko.f
c
c
c
c
c
c
      subroutine GEOKO(x,y,xlat,xlon,i)    ! Urs Kradolfer, 24.3.87
c                         ->         1
c                         <-        -1
c
c Conversion of Swiss-coordinates X,Y [km] to LAT,LON [deg]    i = +1
c Conversion of LAT,LON [deg] to Swiss-coordinates X,Y [km]    i = -1
c
      implicit none
      real x,y,xlat,xlon
      integer i
      real seichmy
      if(.not.(i.eq.1.or.i.eq.-1)) stop'GEOKO>>> specify conversion !!!'
c
      if(i.eq.1)  call EBELL(x,y,xlon,xlat,seichMY)
      if(i.eq.-1) call ELLEB(xlon,xlat,x,y)
c
      end

File: src/sub/geoko.c
void ebell(double yl, double xb, double *l, double *b, double *my);
void elleb(double l, double b, double *x, double *y);

void geoko(double *x, double *y, double xlat, double xlon, int i) {
    double seichmy = 0.0;
    double lat = 0.0;
    double lon = 0.0;

    if (!x || !y) {
        return;
    }
    if (i != 1 && i != -1) {
        return;
    }

    if (i == 1) {
        ebell(xlat, xlon, &lon, &lat, &seichmy);
        *x = lat;
        *y = lon;
    } else {
        elleb(xlon, xlat, x, y);
    }
}
